[8.19] [Detection Engine] Unskip some flaky tests, add better failure messages (#230318)

rylnd opened the pull request:
# Backport

This will backport the following commits from `main` to `8.19`:
 - [[Detection Engine] Unskip some flaky tests, add better failure messages (#230318)](https://github.com/elastic/kibana/pull/230318)

<!--- Backport version: 10.0.1 -->

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport)

<!--BACKPORT [{"author":{"name":"Ryland Herrick","email":"ryalnd@gmail.com"},"sourceCommit":{"committedDate":"2025-08-11T16:40:18Z","message":"[Detection Engine] Unskip some flaky tests, add better failure messages (#230318)\n\n## Summary\n\nWhile the tests affected by this PR are varied, the changes contained\nhere fall under one of two categories:\n\n1. Unskipping flaky tests\n2. Adding better test assertions (in order to produce more actionable\nfailures later)\n\n### Related Issues\n* Closes #224699\n* Closes #224780\n* Closes #220943\n* Closes #202940\n* Closes #202945\n\n\n### Checklist\n\n- [x] [Unit or functional\ntests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)\nwere updated or added to match the most common scenarios","sha":"0a3e7bb41a92b25ca6ae25bef76e53ef634d244b","branchLabelMapping":{"^v9.2.0$":"main","^v(\\d+).(\\d+).\\d+$":"$1.$2"}},"sourcePullRequest":{"labels":["release_note:skip","backport missing","Team:Detection Engine","backport:version","v9.2.0","v9.1.1","v8.19.1"],"title":"[Detection Engine] Unskip some flaky tests, add better failure messages","number":230318,"url":"https://github.com/elastic/kibana/pull/230318","mergeCommit":{"message":"[Detection Engine] Unskip some flaky tests, add better failure messages (#230318)\n\n## Summary\n\nWhile the tests affected by this PR are varied, the changes contained\nhere fall under one of two categories:\n\n1. Unskipping flaky tests\n2. Adding better test assertions (in order to produce more actionable\nfailures later)\n\n### Related Issues\n* Closes #224699\n* Closes #224780\n* Closes #220943\n* Closes #202940\n* Closes #202945\n\n\n### Checklist\n\n- [x] [Unit or functional\ntests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)\nwere updated or added to match the most common scenarios","sha":"0a3e7bb41a92b25ca6ae25bef76e53ef634d244b"}},"sourceBranch":"main","suggestedTargetBranches":["9.1","8.19"],"targetPullRequestStates":[{"branch":"main","label":"v9.2.0","branchLabelMappingKey":"^v9.2.0$","isSourceBranch":true,"state":"MERGED","url":"https://github.com/elastic/kibana/pull/230318","number":230318,"mergeCommit":{"message":"[Detection Engine] Unskip some flaky tests, add better failure messages (#230318)\n\n## Summary\n\nWhile the tests affected by this PR are varied, the changes contained\nhere fall under one of two categories:\n\n1. Unskipping flaky tests\n2. Adding better test assertions (in order to produce more actionable\nfailures later)\n\n### Related Issues\n* Closes #224699\n* Closes #224780\n* Closes #220943\n* Closes #202940\n* Closes #202945\n\n\n### Checklist\n\n- [x] [Unit or functional\ntests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)\nwere updated or added to match the most common scenarios","sha":"0a3e7bb41a92b25ca6ae25bef76e53ef634d244b"}},{"branch":"9.1","label":"v9.1.1","branchLabelMappingKey":"^v(\\d+).(\\d+).\\d+$","isSourceBranch":false,"state":"NOT_CREATED"},{"branch":"8.19","label":"v8.19.1","branchLabelMappingKey":"^v(\\d+).(\\d+).\\d+$","isSourceBranch":false,"state":"NOT_CREATED"}]}] BACKPORT-->

rylnd committed:
[Detection Engine] Unskip some flaky tests, add better failure messages (#230318)

## Summary

While the tests affected by this PR are varied, the changes contained
here fall under one of two categories:

1. Unskipping flaky tests
2. Adding better test assertions (in order to produce more actionable
failures later)

### Related Issues
* Closes #224699
* Closes #224780
* Closes #220943
* Closes #202940
* Closes #202945

### Checklist

- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios

(cherry picked from commit 0a3e7bb41a92b25ca6ae25bef76e53ef634d244b)

# Conflicts:
#	x-pack/test/security_solution_api_integration/test_suites/detections_response/detection_engine/rule_execution_logic/eql/trial_license_complete_tier/eql.ts
#	x-pack/test/security_solution_api_integration/test_suites/detections_response/detection_engine/rule_execution_logic/eql/trial_license_complete_tier/eql_alert_suppression.ts
#	x-pack/test/security_solution_api_integration/test_suites/detections_response/detection_engine/rule_execution_logic/esql/trial_license_complete_tier/esql.ts

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/328914)
* Commit: 839570415ead408372b2d211ace84305ada8bc0a

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/328914#01989ff8-aa88-4dd1-bedb-c50c33de2aba)

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/328914","number":328914,"commit":"839570415ead408372b2d211ace84305ada8bc0a"}],"number":328914}
buildkite-pr-comment-->

rylnd committed:
Fix conflict related to missing backport

These changes are somewhat at odds with what was done in #215757, which
was not backported to 8.19.

rylnd committed:
Merge branch '8.19' into backport/8.19/pr-230318

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/329366)
* Commit: 07124cb644bda309cf8cee505bebf01c5ad82e61

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

### History
* :broken_heart: [Build #328914](https://buildkite.com/elastic/kibana-pull-request/builds/328914) failed 839570415ead408372b2d211ace84305ada8bc0a

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/329366","number":329366,"commit":"07124cb644bda309cf8cee505bebf01c5ad82e61"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/328914","number":328914,"commit":"839570415ead408372b2d211ace84305ada8bc0a"}],"number":329366}
buildkite-pr-comment-->
